### PR TITLE
Modify Dockerfile to include target file

### PR DESCRIPTION
**Pull request by vasireddy99**
**Description:**
This PR is created as a workaround for [ecsobserver extension](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/observer/ecsobserver) to auto-discover Prometheus targets. More details here :  [#5373](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/5373). By creating the file `ecs_sd_targets.yaml` at this path, the collector can auto-discover Prometheus targets and scrape them.

* Also updates w.r.t main branch

**Link to tracking Issue:** #1636 

**Testing:** 

This PR is dependent and to be merged after https://github.com/aws-observability/aws-otel-test-framework/pull/1069

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/PR-build.yml [hunk 1]
```diff
@@ -258,6 +262,19 @@ jobs:
         if: ${{ needs.changes.outputs.changed == 'true' }}
         run: cp -R build aws-otel-collector/build
 
+      - name: Configure AWS Credentials
```
Comment (rapphil) [inline .github/workflows/PR-build.yml:265]: Same here, can you break this in two prs. I assume this is a backport of what is on main. So can you break this in two PRs: one for the changes that are unrelated to the work that you are doing and another for the functionality that you are adding/changing.

File: tools/packaging/windows/aws-otel-collector.wxs [hunk 2]
```diff
@@ -1,13 +1,15 @@
 <?xml version='1.0' encoding='UTF-8'?>
 <Wix xmlns="http://schemas.microsoft.com/wix/2006/wi">
-    <Product Id="*"
+    <Product Id="EBBD8548-75D1-41D3-A402-ABE189F0C167"
```
Comment (rapphil) [inline tools/packaging/windows/aws-otel-collector.wxs:3]: This also seems unrelated.

File: cmd/awscollector/Dockerfile [hunk 3]
```diff
@@ -72,6 +75,7 @@ COPY --from=certs /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certifica
 COPY --from=package /workspace/awscollector /awscollector
 COPY --from=package /workspace/config/ /etc/
 COPY --from=package /workspace/healthcheck /healthcheck
+COPY --from=ecstarget ecs_sd_targets.yaml /etc/
```
Comment (rapphil) [inline cmd/awscollector/Dockerfile:78]: can you detail how adding an empty file here will create an workaround? also can you add comments describing what is the workaround and a link to the upstream issue?

your idea is to make this test work only in the dev branch?

Comment (rapphil) [reply cmd/awscollector/Dockerfile:78]: by comments I mean that you should add comments in the dockerfile explaining that this line is a workaround and when it could be removed.

Comment (vasireddy99) [reply cmd/awscollector/Dockerfile:78]: yeah I am currently not sure if there is going to be any fix for this, probably the solution that we have?

Comment (rapphil) [reply cmd/awscollector/Dockerfile:78]: Sure, but what is special about copying an empty file in a docker image? can you add comments in the dockerfile, since this is a workaround?

**Comment by vasireddy99** [comment]
clsoing this pr infavor of https://github.com/aws-observability/aws-otel-collector/pull/1828 . Since, this branch is created with main as base.
